using_restricted_array.rb

```diff
@@ -22,7 +22,7 @@ def print_array(array)
     printed_array << array[i]
     i += 1
   end
-  print printed_array
+  print printed_array # if I add a return here, it does not print the array. Why?
```

Comment:
If you add a return after the print line, it should print the array. If it's not, let's explore further together.

Reply:
If you just do a `print array[i]` instead of adding to a new array, you wouldn't incur the space complexity. I'm curious, what do you gain by using the printed_array? Does it simplify anything?